test/tools/router_check/router.cc

```diff
@@ -91,6 +105,25 @@ void RouterCheckTool::assignRuntimeFraction(
   }
 }
 
+void RouterCheckTool::finalizeHeaders(ToolConfig& tool_config,
+                                      Envoy::StreamInfo::StreamInfoImpl stream_info) {
+  if (!headers_finalized_ && tool_config.route_ != nullptr) {
+    if (tool_config.route_->directResponseEntry() != nullptr) {
+      tool_config.route_->directResponseEntry()->rewritePathHeader(*tool_config.request_headers_,
+                                                                   true);
```

Comment:
Is this explicitly needed? I don't think this really makes sense for direct responses?

Reply:
it's needed for redirects - I added it in a prior fix in https://github.com/envoyproxy/envoy/pull/9775

Reply:
Ah OK, got it. Thanks.